packages/eas-cli/src/credentials/credentialsJson/update.ts

```diff
@@ -293,13 +294,10 @@ async function updateFileAsync(
 }
 
 async function isFileUntrackedAsync(path: string): Promise<boolean> {
-  const withUntrackedFiles = await gitStatusAsync({ showUntracked: true });
-  const trackedFiles = await gitStatusAsync({ showUntracked: false });
-  const pathWithoutLeadingDot = path.replace(/^\.\//, ''); // remove leading './' from path
-  return (
-    withUntrackedFiles.includes(pathWithoutLeadingDot) &&
-    !trackedFiles.includes(pathWithoutLeadingDot)
-  );
+  if (vcs instanceof GitClient) {
+    return await vcs.isFileUntrackedAsync(path);
+  }
+  return false;
```

Comment:
Hmm, maybe instead of checking if the object is an instance of a particular class, clients that don't support tracking files should always return `false` from `isFileUntrackedAsync`?

Reply:
... and I think it's already implemented this way. I think we can skip the `if` entirely.

Reply:
that was a bug it should return true by default which would not be correct behavior for `isFileUntrackedAsync`, that is why I did not add that method to the common interface.

Reply:
I don't want to add that to the common interface, it's git specific terminology plus we are using it in a bit unexpected way, so it might be confusing for someone implementing the interface